File: Examples/test-suite/lua/exception_memory_leak_runme.lua
eml=require("exception_memory_leak")
catch_undef_globs() -- catch "undefined" global variables

a = eml.Foo()
assert(eml.Foo_get_count() == 1)
b = eml.Foo()
assert(eml.Foo_get_count() == 2)

-- Normal behaviour
eml.trigger_internal_swig_exception("no problem", a)
assert(eml.Foo_get_count() == 2)
assert(eml.Foo_get_freearg_count() == 1)

-- SWIG exception triggered and handled (return new object case)
assert(not pcall(eml.trigger_internal_swig_exception, "null", b))
assert(eml.Foo_get_count() == 2)
assert(eml.Foo_get_freearg_count() == 2)

-- SWIG exception triggered and handled (return by value case).
assert(not pcall(eml.trigger_internal_swig_exception, "null"))
assert(eml.Foo_get_count() == 2)

File: Examples/test-suite/go/exception_memory_leak_runme.go
package main

import "strings"
import . "swigtests/exception_memory_leak"

func main() {
	a := NewFoo()
	if FooGet_count() != 1 {
		panic("Foo count not 1")
	}
	b := NewFoo()
	if FooGet_count() != 2 {
		panic("Foo count not 2")
	}

	// Normal behaviour
	Trigger_internal_swig_exception("no problem", a)
	if FooGet_count() != 2 {
		panic("Foo count not 2")
	}
	if FooGet_freearg_count() != 1 {
		panic("Foo count not 1")
	}

	// SWIG exception triggered and handled (return new object case)
	func() {
		defer func() {
			e := recover()
			if strings.Index(e.(string), "Let's") != 0 {
				panic(e.(string))
			}
		}()
		Trigger_internal_swig_exception("null", b)
		panic("Trigger_internal_swig_exception didn't panic")
	}()
	if FooGet_count() != 2 {
		panic("Foo count not 2")
	}
	if FooGet_freearg_count() != 2 {
		panic("Foo count not 2")
	}

	// SWIG exception triggered and handled (return by value case).
	func() {
		defer func() {
			e := recover()
			if strings.Index(e.(string), "Let's") != 0 {
				panic(e.(string))
			}
		}()
		Trigger_internal_swig_exception("null")
		panic("Trigger_internal_swig_exception didn't panic")
	}()
	if FooGet_count() != 2 {
		panic("Foo count not 2")
	}
}
